server/app/src/androidTest/java/androidx/test/uiautomator/UiScrollableCustom.java

```diff
@@ -21,4 +21,30 @@ public boolean scrollIntoView(UiSelector selector) throws UiObjectNotFoundExcept
         return found;
     }
 
+    public boolean scrollIntoView(UiSelector selector, int index) throws UiObjectNotFoundException {
+        Tracer.trace(new Object[]{selector});
+        this.setMaxSearchSwipes(1000);
+        if (this.exists(selector.index(index))) {
+            return true;
+        } else {
+            this.flingToBeginning(getMaxSearchSwipes());
+            if (this.exists(selector.index(index))) {
+                return true;
+            } else {
+                for(int x = 0; x < getMaxSearchSwipes(); ++x) {
```

Comment:
Though the iteration is trying to find the element at the index position, The iteration variable x is not related to the index. From built-in Uiscrolling methods, I notice the "x" variable is mostly used for scrolling and to maintain the consistence it would be nice to use the same.